diff --git a/src/miner.cpp b/src/miner.cpp
index adfedcc25..e3d2f3479 100644
--- a/src/miner.cpp
+++ b/src/miner.cpp
@@ -58,6 +58,8 @@ int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParam
     if (consensusParams.fPowAllowMinDifficultyBlocks)
         pblock->nBits = GetNextWorkRequired(pindexPrev, pblock, consensusParams);
 
+    pblock->nTime += 10 * 60;
+
     return nNewTime - nOldTime;
 }
 
