diff --git a/src/util.cpp b/src/util.cpp
index 1908a6b13..8c95a9559 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -525,13 +525,13 @@ void PrintExceptionContinue(const std::exception* pex, const char* pszThread)
 
 fs::path GetDefaultDataDir()
 {
-    // Windows < Vista: C:\Documents and Settings\Username\Application Data\BitZenyPlus
-    // Windows >= Vista: C:\Users\Username\AppData\Roaming\BitZenyPlus
-    // Mac: ~/Library/Application Support/BitZenyPlus
-    // Unix: ~/.bitzenyplus
+    // Windows < Vista: C:\Documents and Settings\Username\Application Data\Bitcoin
+    // Windows >= Vista: C:\Users\Username\AppData\Roaming\Bitcoin
+    // Mac: ~/Library/Application Support/Bitcoin
+    // Unix: ~/.bitcoin
 #ifdef WIN32
     // Windows
-    return GetSpecialFolderPath(CSIDL_APPDATA) / "BitZenyPlus";
+    return GetSpecialFolderPath(CSIDL_APPDATA) / "BitZeny";
 #else
     fs::path pathRet;
     char* pszHome = getenv("HOME");
@@ -541,10 +541,10 @@ fs::path GetDefaultDataDir()
         pathRet = fs::path(pszHome);
 #ifdef MAC_OSX
     // Mac
-    return pathRet / "Library/Application Support/BitZenyPlus";
+    return pathRet / "Library/Application Support/BitZeny";
 #else
     // Unix
-    return pathRet / ".bitzenyplus";
+    return pathRet / ".bitzeny";
 #endif
 #endif
 }
